### ISSUE TITLE
makeps3iso compiled against glibc version not available in Ubuntu LTS/KDE Neon

### ISSUE THREAD

**Issue by xdpirate** [Klairm/LINUX-PS3-ISO-TOOLS#3]
Hi - tried to use this tonight to avoid having to transfer my PS3 game to a Windows PC, but the UI throws an error once I've selected my folders:

![image](https://user-images.githubusercontent.com/1757462/181850786-65a2ba46-033f-4592-b0c8-9e1a3189b166.png)

```./bin/makeps3iso: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by ./bin/makeps3iso)```

Checking Ubuntu LTS repositories:
```
/opt/linux-ps3-iso-tools$ apt list libc6
Listing... Done
libc6/focal-updates,now 2.31-0ubuntu9.9 amd64 [installed]
libc6/focal-updates,now 2.31-0ubuntu9.9 i386 [installed,automatic]
```

It seems `makeps3iso` is compiled against a newer version of `glibc` than is included in the Ubuntu LTS repositories, where the newest version is 2.31. Is there any way to rectify this? I am running KDE neon user edition, which is currently using Ubuntu 20.04 LTS repositories for anything non-plasma.

**Comment by Klairm** [comment]
> Hi - tried to use this tonight to avoid having to transfer my PS3 game to a Windows PC, but the UI throws an error once I've selected my folders:
> 
> ![image](https://user-images.githubusercontent.com/1757462/181850786-65a2ba46-033f-4592-b0c8-9e1a3189b166.png)
> 
> `` ./bin/makeps3iso: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by ./bin/makeps3iso) ``
> 
> Checking Ubuntu LTS repositories:
> 
> ```
> /opt/linux-ps3-iso-tools$ apt list libc6
> Listing... Done
> libc6/focal-updates,now 2.31-0ubuntu9.9 amd64 [installed]
> libc6/focal-updates,now 2.31-0ubuntu9.9 i386 [installed,automatic]
> ```
> 
> It seems `makeps3iso` is compiled against a newer version of `glibc` than is included in the Ubuntu LTS repositories, where the newest version is 2.31. Is there any way to rectify this? I am running KDE neon user edition, which is currently using Ubuntu 20.04 LTS repositories for anything non-plasma.

You could either upgrade your Ubuntu version , 22.04 is LTS too! Or the other workaroud would be download the required version of glibc and do a link to that file.

You could also try to compile the tools .c files with the Makefile and it should use the glibc version you're using, if you can't do this if you give me some time I'll try to create a docker and compile the binaries for that version

**Comment by xdpirate** [comment]
Thanks for your swift reply, I actually just realized the source for `makeps3iso` was included, so I ran make and tried to compile it. It compiled (albeit with a bunch of warnings), then I replaced the binary in `bin` with the newly created binary, and re-ran the UI - it seems to have worked perfectly and created an ISO of the expected size.

KDE neon has unfortunately not yet rebased on 22.04, but I've been told that will happen at some point around one of the upcoming 22.04 LTS point releases :) Thanks!
